registration/static/js/date-entry.js

```diff
@@ -38,14 +38,11 @@ $(document).ready(function (e) {
         return;
     }
     let [year, month, day] = dob.split('-');
-    if (year) {
+    if (year && month && day) {
         $("#byear").val(year);
         $("#bmonth").val(month);
         $('#bmonth').change();
         $("#bday").val(day);
-    } else {
-        day = $('#bday').val();
-        $('#bmonth').change();
-        $('#bday').val(day);
```

Comment:
@rechner Do you remember why this `if (year)` check exists? I couldn't immediately figure out what it's doing. Let me know if I need to add it back.

Reply:
Oh probably nothing good.  I imagine it's just in case dob is unset or malformed.